apps/dashboard/src/app/(main)/(protected)/projects/[projectId]/metrics/line-chart.tsx

```diff
@@ -57,48 +57,105 @@ export function LineChartDisplay({
   );
 }
 
-export function MiniLineChartDisplay({
-  config, datapoints
+const BRAND_CONFIG = {
+  email: {
+    label: 'Email',
+    color: '#ffffff'
+  },
+  magiclink: {
+    label: 'Magic Link',
+    color: '#A657F0'
+  },
+  passkey: {
+    label: 'Passkey',
+    color: '#D2B6EF'
+  },
+  google: {
+    label: 'Google',
+    color: '#4285F4',
+  },
+  github: {
+    label: 'GitHub',
+    color: '#238636',
+  },
+  microsoft: {
+    label: 'Microsoft',
+    color: '#F35325',
+  },
+  spotify: {
+    label: 'Spotify',
+    color: '#1ED760'
+  },
+  facebook: {
+    label: 'Facebook',
+    color: '#0866FF',
+  },
+  discord: {
+    label: 'Discord',
+    color: '#5865F2',
+  },
+  gitlab: {
+    label: 'GitLab',
+    color: '#FC6D26'
+  },
+  bitbucket: {
+    label: 'Bitbucket',
+    color: '#0052CC',
+  },
+  linkedin: {
+    label: 'LinkedIn',
+    color: '#0A66C2',
+  },
+  apple: {
+    label: 'Apple',
+    color: '#F47CAD',
+  },
+  x: {
+    label: 'X (Twitter)',
+    color: '#1D9BF0',
+  }
+};
+
+export type AuthMethodDatapoint = {
+  method: keyof typeof BRAND_CONFIG,
+  count: number,
+};
+
+export function DonutChartDisplay({
+  datapoints
 }: {
-  config: LineChartDisplayConfig,
-  datapoints: DataPoint[],
+  datapoints: AuthMethodDatapoint[],
 }) {
   return (
-    <Card className="max-w-[300px]">
+    <Card>
       <CardHeader>
-        <CardTitle className="text-3xl">
-          {datapoints[datapoints.length - 1].activity}
+        <CardTitle>
+          Login Methods
         </CardTitle>
-        <CardDescription className="text-lg">
-          {config.name}
+        <CardDescription>
+          Login methods used
         </CardDescription>
       </CardHeader>
       <CardContent>
-        <ChartContainer config={config.chart} className='w-full max-h-[300px] p-4'>
-          <AreaChart accessibilityLayer data={datapoints}>
+        <ChartContainer config={BRAND_CONFIG} className='w-full max-h-[300px] p-4'>
```

Comment:
what happens when the login method isn't in here? eg. we add a new one, but forget to add it here

Reply:
Just tested: it gets classified as "other"